<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Outlier Detection in Genetic Data • molic</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Outlier Detection in Genetic Data">
<meta property="og:description" content="molic">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">molic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/dermatitis.html">Detecting Skin Diseases</a>
    </li>
    <li>
      <a href="../articles/genetic_example.html">Outlier Detection in Genetic Data</a>
    </li>
    <li>
      <a href="../articles/outlier_intro.html">The Outlier Model</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mlindsk/molic/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Outlier Detection in Genetic Data</h1>
                        <h4 class="author">Mads Lindskou</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mlindsk/molic/blob/master/vignettes/genetic_example.Rmd"><code>vignettes/genetic_example.Rmd</code></a></small>
      <div class="hidden name"><code>genetic_example.Rmd</code></div>

    </div>

    
    
<div id="the-1000-genomes-project-data" class="section level1">
<h1 class="hasAnchor">
<a href="#the-1000-genomes-project-data" class="anchor"></a>The 1000 Genomes Project Data</h1>
<p>The outlier detection method <span class="citation">(Lindskou, Svante Eriksen, and Tvedebrink 2019)</span> arose from a problem in the forensic science community where it is of great interest to make statements about the geographical origin of a DNA sample. This is in general a very complicated matter. More over, when DNA markers are in linkage disequilibrium things get even more complicated. The 1000 Genomes Project set out to provide a comprehensive description of common human genetic variation by applying whole-genome sequencing to a diverse set of individuals from multiple populations <span class="citation">(The 1000 Genomes Project Consortium 2015)</span>. In the <strong>molic</strong> package we include the final data from the project which includes <span class="math inline">\(2504\)</span> DNA profiles coming from five different continental regions; Europe (EUR), Africa (AFR), America (AMR), East Asia (EAS), South Asia (SAS). Each DNA profile is the compound of two rows in the data <code>tgp_dat</code>, one for each chromosome copy. This makes the outlier procedure slightly more complicated since we must fit a model to each copy and aggregate the information (but we shall see in a moment that it is not hard to do using the <strong>molic</strong> package).</p>
<p>The data includes <span class="math inline">\(276\)</span> SNP markers grouped in <span class="math inline">\(97\)</span> microhaplotypes; in other words, the SNPs within a microhaplotype are so close that they cannot be assumed to be in linkage disequilibrium and we must take into account their mutual dependencies. We do this with the use of decomposable graphical models. See the <code>outlier_intro</code> for more details on the model. The microhaplotype grouping is provided is the list <code>tgp_haps</code>.</p>
<div id="setup-and-data" class="section level2">
<h2 class="hasAnchor">
<a href="#setup-and-data" class="anchor"></a>Setup and Data</h2>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">molic</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ess</span>)   <span class="co"># for fit_components</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">7</span>)    <span class="co"># for reproducibility</span>

<span class="co">## Data</span>

<span class="co"># The components - here microhaplotypes</span>
<span class="no">haps</span> <span class="kw">&lt;-</span> <span class="no">tgp_haps</span>

<span class="co"># All the Europeans</span>
<span class="no">eur</span> <span class="kw">&lt;-</span> <span class="no">tgp_dat</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">pop_meta</span> <span class="kw">==</span> <span class="st">"EUR"</span>)

<span class="co"># Extracting the two databases for each copy of the chromosomes</span>
<span class="no">eur_a</span> <span class="kw">&lt;-</span> <span class="no">eur</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"a$"</span>, <span class="no">sample_name</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(-<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>:<span class="fl">2</span>))

<span class="no">eur_b</span> <span class="kw">&lt;-</span> <span class="no">eur</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"b$"</span>, <span class="no">sample_name</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(-<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>:<span class="fl">2</span>))</pre></body></html></div>
<p>We peak at data:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">eur</span>, <span class="kw">n_extra</span> <span class="kw">=</span> <span class="fl">0</span>)</pre></body></html></div>
<pre><code>## # A tibble: 1,006 x 304
##    sample_name pop_meta rs28529526 rs10134526 rs260694 rs11123719 rs11691107
##    &lt;chr&gt;       &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;     
##  1 HG00096a    EUR      G          C          T        T          C         
##  2 HG00097a    EUR      G          T          T        C          C         
##  3 HG00099a    EUR      G          C          T        T          C         
##  4 HG00100a    EUR      G          C          T        T          C         
##  5 HG00101a    EUR      G          C          T        T          C         
##  6 HG00102a    EUR      G          C          T        T          C         
##  7 HG00103a    EUR      G          C          T        C          C         
##  8 HG00105a    EUR      G          T          T        C          C         
##  9 HG00106a    EUR      G          C          T        C          C         
## 10 HG00107a    EUR      G          C          T        T          C         
## # … with 996 more rows, and 297 more variables</code></pre>
<p>We now fit the interaction graphs for each chromosome copy. We use <code>fit_components</code> since we know, that SNPs between haplotypes are independent. This also makes the procedure much faster.</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">ga</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ess/man/fit_components.html">fit_components</a></span>(<span class="no">eur_a</span>, <span class="kw">comp</span> <span class="kw">=</span> <span class="no">haps</span>, <span class="kw">trace</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">gb</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ess/man/fit_components.html">fit_components</a></span>(<span class="no">eur_b</span>, <span class="kw">comp</span> <span class="kw">=</span> <span class="no">haps</span>, <span class="kw">trace</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">ga</span>)</pre></body></html></div>
<pre><code>##  A Decomposable Graph With 
##  ------------------------- 
##   Nodes: 302 
##   Edges: 234 / 45451 
##   Cliques: 161 
##    - max: 4 
##    - min: 1 
##    - avg: 2.18 
##   &lt;gengraph, list&gt; 
##  -------------------------</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">ga</span>, <span class="kw">vertex.size</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">vertex.label</span> <span class="kw">=</span> <span class="fl">NA</span>)</pre></body></html></div>
<p><img src="genetic_example_files/figure-html/unnamed-chunk-2-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div id="example-1---testing-which-observations-within-data-are-outliers" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1---testing-which-observations-within-data-are-outliers" class="anchor"></a>Example 1 - Testing which observations within data are outliers</h2>
<p>Fitting the model:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="co"># Only 500 simulations is used here to exeplify</span>
<span class="co"># The default number of simulations is 10,000</span>
<span class="no">m1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_outlier.html">fit_outlier</a></span>(<span class="no">eur_a</span>, <span class="no">ga</span>, <span class="kw">nsim</span> <span class="kw">=</span> <span class="fl">500</span>) <span class="co"># consider using more cores (ncores argument)</span>
<span class="no">m2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_outlier.html">fit_outlier</a></span>(<span class="no">eur_b</span>, <span class="no">gb</span>, <span class="kw">nsim</span> <span class="kw">=</span> <span class="fl">500</span>) <span class="co"># consider using more cores (ncores argument)</span>
<span class="no">m</span>  <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_mixed_outlier.html">fit_mixed_outlier</a></span>(<span class="no">m1</span>, <span class="no">m2</span>)</pre></body></html></div>
<p>Model information:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">m</span>)</pre></body></html></div>
<pre><code>## 
##  -------------------------------- 
##   Simulations: 500 
##   Variables: 302 
##   Observations: 503 
##   Estimated mean: 482.36 
##   Estimated variance: 345.85 
##  --------------------------------
##   Critical value: 516.8367 
##   Alpha: 0.05 
##   &lt;mixed_outlier, outlier, outlier_model, list&gt; 
##  --------------------------------</code></pre>
<p>The distribution of the test statistic:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">m</span>)</pre></body></html></div>
<p><img src="genetic_example_files/figure-html/unnamed-chunk-5-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Locating the outliers:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">outs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/outliers.html">outliers</a></span>(<span class="no">m</span>)
<span class="no">eur_a_outs</span> <span class="kw">&lt;-</span> <span class="no">eur_a</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">outs</span>), ]
<span class="no">eur_b_outs</span> <span class="kw">&lt;-</span> <span class="no">eur_b</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">outs</span>), ]
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">eur_a_outs</span>, <span class="kw">n_extra</span> <span class="kw">=</span> <span class="fl">0</span>)</pre></body></html></div>
<pre><code>## # A tibble: 19 x 302
##    rs28529526 rs10134526 rs260694 rs11123719 rs11691107 rs12469721 rs3101043
##    &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;    
##  1 G          C          T        T          C          A          C        
##  2 G          C          T        C          C          T          T        
##  3 G          C          T        T          C          T          T        
##  4 G          C          T        C          C          A          T        
##  5 G          T          T        T          C          A          T        
##  6 G          C          T        C          C          T          T        
##  7 G          T          T        T          C          A          C        
##  8 G          C          T        C          C          A          C        
##  9 G          C          T        T          C          A          C        
## 10 G          C          T        C          C          T          C        
## 11 G          C          T        T          C          T          C        
## 12 G          C          T        C          C          A          T        
## 13 G          C          T        T          C          T          T        
## 14 G          C          T        T          T          T          T        
## 15 G          C          T        T          C          A          C        
## 16 G          C          T        T          C          T          T        
## 17 G          C          T        T          C          T          C        
## 18 G          T          T        T          C          T          C        
## 19 G          T          T        T          C          A          T        
## # … with 295 more variables</code></pre>
<p>Retrieving the deviance test statistic for individual observations:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">x1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="no">eur_a_outs</span>[<span class="fl">1</span>, ], <span class="no">eur_b_outs</span>[<span class="fl">1</span>, ])
<span class="no">x2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="no">eur_a</span>[<span class="fl">1</span>, ], <span class="no">eur_b</span>[<span class="fl">1</span>, ])
<span class="no">dev1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/deviance.html">deviance</a></span>(<span class="no">m</span>, <span class="no">x1</span>) <span class="co"># falls within the critical region in the plot (the red area)</span>
<span class="no">dev2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/deviance.html">deviance</a></span>(<span class="no">m</span>, <span class="no">x2</span>) <span class="co"># falls within the acceptable region in the plot</span>

<span class="no">dev1</span></pre></body></html></div>
<pre><code>## [1] 537.9733</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="no">dev2</span></pre></body></html></div>
<pre><code>## [1] 496.799</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="co"># Retrieving the pvalues</span>
<span class="fu"><a href="../reference/pval.html">pval</a></span>(<span class="no">m</span>, <span class="no">dev1</span>)</pre></body></html></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="../reference/pval.html">pval</a></span>(<span class="no">m</span>, <span class="no">dev2</span>)</pre></body></html></div>
<pre><code>## [1] 0.208</code></pre>
</div>
<div id="example-2---testing-if-a-new-observation-is-an-outlier" class="section level2">
<h2 class="hasAnchor">
<a href="#example-2---testing-if-a-new-observation-is-an-outlier" class="anchor"></a>Example 2 - Testing if a new observation is an outlier</h2>
<p>Testing if an American is an outlier in Europe:</p>
<div class="sourceCode"><html><body><pre class="r"><span class="no">amr</span> <span class="kw">&lt;-</span> <span class="no">tgp_dat</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">pop_meta</span> <span class="kw">==</span> <span class="st">"AMR"</span>)

<span class="no">z1</span>  <span class="kw">&lt;-</span> <span class="no">amr</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"a$"</span>, <span class="no">sample_name</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">haps</span>))) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="fl">1</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>()

<span class="no">z2</span>  <span class="kw">&lt;-</span> <span class="no">amr</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"b$"</span>, <span class="no">sample_name</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unname.html">unname</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="no">haps</span>))) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span>(<span class="fl">1</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>()

<span class="co"># Only 500 simulations is used here to exemplify</span>
<span class="co"># The default number of simulations is 10,000</span>
<span class="no">m3</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_outlier.html">fit_outlier</a></span>(<span class="no">eur_a</span>, <span class="no">ga</span>, <span class="no">z1</span>, <span class="kw">nsim</span> <span class="kw">=</span> <span class="fl">500</span>) <span class="co"># consider using more cores (ncores argument)</span>
<span class="no">m4</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_outlier.html">fit_outlier</a></span>(<span class="no">eur_b</span>, <span class="no">gb</span>, <span class="no">z2</span>, <span class="kw">nsim</span> <span class="kw">=</span> <span class="fl">500</span>) <span class="co"># consider using more cores (ncores argument)</span>
<span class="no">m5</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_mixed_outlier.html">fit_mixed_outlier</a></span>(<span class="no">m3</span>, <span class="no">m4</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">m5</span>)</pre></body></html></div>
<pre><code>## 
##  -------------------------------- 
##   Simulations: 500 
##   Variables: 302 
##   Observations: 504 
##   Estimated mean: 483.14 
##   Estimated variance: 341.9 
##  --------------------------------
##   Critical value: 516.6007 
##   Deviance: 508.0499 
##   P-value: 0.098 
##   Alpha: 0.05 
##   &lt;mixed_outlier, novelty, outlier_model, list&gt; 
##  --------------------------------</code></pre>
<div class="sourceCode"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="no">m5</span>)</pre></body></html></div>
<p><img src="genetic_example_files/figure-html/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;"></p>
<p>Thus, <code>z</code> does not deviate significantly from the Europe database. The red area depicts the <span class="math inline">\(0.05\)</span> significance level and the dotted line is the observed deviance of <code>z</code> (508.0498712). This is not a complete surprise, since Americans and Europeans have many similarities.</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-lindskououtlier">
<p>Lindskou, Mads, Poul Svante Eriksen, and Torben Tvedebrink. 2019. “Outlier Detection in Contingency Tables Using Decomposable Graphical Models.” <em>Scandinavian Journal of Statistics</em>. Wiley Online Library. doi:<a href="https://doi.org/10.1111/sjos.12407">10.1111/sjos.12407</a>.</p>
</div>
<div id="ref-10002015global">
<p>The 1000 Genomes Project Consortium. 2015. “A Global Reference for Human Genetic Variation.” <em>Nature</em> 526 (7571). Nature Publishing Group: 68. doi:<a href="https://doi.org/10.1038/nature15393">10.1038/nature15393</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mads Lindskou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
